CMakeLists.txt

```diff
@@ -1,44 +1,95 @@
-# This file is only used to enable the building librealsense
-# as a debian package for use with ROS (http://ros.org)
-# via catkin_make (which invokes cmake)
+cmake_minimum_required(VERSION 3.1)
 
-cmake_minimum_required(VERSION 2.8.3)
+project(Realsense)
 
-project(librealsense)
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/CMake)
 
-include(ExternalProject)
-
-find_package(catkin REQUIRED)
-
-externalproject_add(${PROJECT_NAME}_extern
-  PREFIX "external_proj"
+set(REALSENSE_CPP 
+    src/context.cpp
+    src/device.cpp
+    src/f200-private.cpp
+    src/f200.cpp
+    src/image.cpp
+    src/log.cpp
+    src/r200-private.cpp
+    src/r200.cpp
+    src/rs.cpp
+    src/stream.cpp
+    src/sync.cpp
+    src/types.cpp
+    src/uvc-libuvc.cpp
+    src/uvc-v4l2.cpp
+    src/uvc-wmf.cpp
+    src/uvc.cpp
+    src/verify.c
+)
 
-  SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}
+set(REALSENSE_HPP 
+    src/context.h
+    src/device.h
+    src/f200-private.h
+    src/f200.h
+    src/image.h
+    src/r200-private.h
+    src/r200.h
+    src/stream.h
+    src/sync.h
+    src/types.h
+    src/uvc.h
+)
 
-  CONFIGURE_COMMAND echo "No Configuration Step"
+if(WIN32)
+    set(BACKEND RS_USE_WMF_BACKEND)
+    set(REALSENSE_DEF CMake/realsense.def)
+elseif(APPLE)
+    set(BACKEND RS_USE_LIBUVC_BACKEND)
+else()
+    set(BACKEND RS_USE_V4L2_BACKEND)
+endif()
+add_definitions(-D${BACKEND} -DUNICODE)
 
-  BUILD_IN_SOURCE 1
-  BUILD_COMMAND $(MAKE) lib/${PROJECT_NAME}.so
+if(UNIX)
+    list(APPEND REALSENSE_CPP
+        src/libuvc/ctrl.c
+        src/libuvc/dev.c
+        src/libuvc/diag.c
+        src/libuvc/frame.c
+        src/libuvc/init.c
+        src/libuvc/stream.c
+    )
+    list(APPEND REALSENSE_HPP
+        src/libuvc/libuvc_config.h
+        src/libuvc/libuvc.h
+        src/libuvc/libuvc_internal.h
+        src/libuvc/utlist.h
+    )
+    
+    find_package(LibUSB REQUIRED)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC -pedantic -mssse3 -Ofast")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-switch -Wno-multichar")
+endif()
 
-  INSTALL_DIR ${CATKIN_DEVEL_PREFIX}
-  INSTALL_COMMAND cp <SOURCE_DIR>/lib/${PROJECT_NAME}.so <INSTALL_DIR>/lib/ COMMAND cp -r <SOURCE_DIR>/include/${PROJECT_NAME} <INSTALL_DIR>/include/
-)
+option(BUILD_SHARED_LIBS "Build shared library" ON)
+if(BUILD_SHARED_LIBS)
+    add_library(realsense SHARED ${REALSENSE_CPP} ${REALSENSE_HPP} ${REALSENSE_DEF})
+    target_link_libraries(realsense ${LibUSB_LIBRARIES})
+else()
+    add_library(realsense STATIC ${REALSENSE_CPP} ${REALSENSE_HPP})
```

Comment:
Thank you for supports to generate static link library.
Currently, The name of static link library is <code>realsense-s</code> on Windows.
I think that the suffix (<code>-s</code>) in the same way as currently should be added when generate static link library by using CMake. (Or, should change this specification?)
What do you think?
Thanks,